clients/client-core/src/client/real_messages_control/acknowledgement_control/retransmission_request_listener.rs

```diff
@@ -121,9 +126,17 @@ where
 
     pub(super) async fn run(&mut self) {
         debug!("Started RetransmissionRequestListener");
-        while let Some(timed_out_ack) = self.request_receiver.next().await {
-            self.on_retransmission_request(timed_out_ack).await;
+
+        while !self.shutdown.is_shutdown() {
+            tokio::select! {
+                Some(timed_out_ack) = self.request_receiver.next() => {
```

Comment:
actually what would be the behaviour in this `select` if value returned was `None`? would it still get caught?